clang/lib/Sema/SemaType.cpp

```diff
@@ -2306,7 +2306,7 @@ QualType Sema::BuildBitIntType(bool IsUnsigned, Expr *BitWidth,
   }
 
   const TargetInfo &TI = getASTContext().getTargetInfo();
-  if (NumBits > TI.getMaxBitIntWidth()) {
+  if (NumBits > TI.getMaxBitIntWidth() && !Context.getLangOpts().IntelFPGA) {
```

Comment:
Shouldn't there be some code handling the larger size for this target? Is just removing the error enough?

Reply:
I believe it is enough.  The actual handling is done by the backend or via library calls.

Reply:
Actually, this is not enough.  We need to make sure we don't emit anything too large for the LLVM backend.

IMO, the RIGHT place for this is probably to get the proper target-info for the FPGA to change its answer to 'getMaxBitIntWidth' based on the FPGA status to the LLVM max value.

Reply:
> Actually, this is not enough. We need to make sure we don't emit anything too large for the LLVM backend.
> 
> IMO, the RIGHT place for this is probably to get the proper target-info for the FPGA to change its answer to 'getMaxBitIntWidth' based on the FPGA status to the LLVM max value.

If we go with this way in any following patch, 2048 is a reasonable upper-bound limit for the FPGA target.